model/config.go

```diff
@@ -44,13 +45,24 @@ func NewCedarConfig(env cedar.Environment) *CedarConfig {
 }
 
 var (
-	cedarConfigurationIDKey      = bsonutil.MustHaveTag(CedarConfig{}, "ID")
-	cedarConfigurationSplunkKey  = bsonutil.MustHaveTag(CedarConfig{}, "Splunk")
-	cedarConfigurationSlackKey   = bsonutil.MustHaveTag(CedarConfig{}, "Slack")
-	cedarConfigurationLDAPKey    = bsonutil.MustHaveTag(CedarConfig{}, "LDAP")
-	cedarConfigurationCAKey      = bsonutil.MustHaveTag(CedarConfig{}, "CA")
-	cedarConfigurationFlagsKey   = bsonutil.MustHaveTag(CedarConfig{}, "Flags")
-	cedarConfigurationServiceKey = bsonutil.MustHaveTag(CedarConfig{}, "Service")
+	cedarConfigurationIDKey             = bsonutil.MustHaveTag(CedarConfig{}, "ID")
+	cedarConfigurationSplunkKey         = bsonutil.MustHaveTag(CedarConfig{}, "Splunk")
+	cedarConfigurationSlackKey          = bsonutil.MustHaveTag(CedarConfig{}, "Slack")
+	cedarConfigurationLDAPKey           = bsonutil.MustHaveTag(CedarConfig{}, "LDAP")
+	cedarConfigurationCAKey             = bsonutil.MustHaveTag(CedarConfig{}, "CA")
+	cedarConfigurationFlagsKey          = bsonutil.MustHaveTag(CedarConfig{}, "Flags")
+	cedarConfigurationServiceKey        = bsonutil.MustHaveTag(CedarConfig{}, "Service")
+	cedarConfigurationChangeDetectorKey = bsonutil.MustHaveTag(CedarConfig{}, "ChangeDetector")
+)
+
+type ChangeDetectorConfig struct {
```

Comment:
should we add an implementation name/string?

Reply:
I think I did what you were asking for, let me know if I misinterpreted what you meant.